### PR TITLE
mac build support

### PR DESCRIPTION
**Pull request by fprimex**
Don't know if you want to actually merge this, but I have one other in-progress part to this that tries some things with `install-name-tool` and other stuff. I can push that here and you can maybe cherry pick.

The bottom line is that I can compile fine but I can't get all of the dylibs to go into the bundle correctly for zip file distribution to computers that don't have Qt and other libraries installed. So, for people who want to brew install, compile, and run this will work, but of course I want to fix the bundle.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -59,31 +82,18 @@ if [[ $1 != "rpi3" ]]; then
     fi
     make -j4 release
     cp $base_dir/mupen64plus-gui/build/release/mupen64plus-gui.exe $install_dir
+  elif [[ $UNAME == "Darwin" ]]; then
+    /usr/local/Cellar/qt5/*/bin/qmake ../mupen64plus-gui.pro
+    make -j4
+    mv $install_dir/* $base_dir/mupen64plus-gui/build/mupen64plus-gui.app/Contents/MacOS
```
Comment (loganmc10) [inline build.sh]: This isn't how things should work. mupen64plus-gui is independent from the other components (mupen64plus-core, etc...). Perhaps looking at m64py will give some hints as to how they install things?

Comment (fprimex) [inline build.sh]: I'll push part 2 which is certainly an improvement, yet still does not work, shortly.

**Comment by fprimex** [comment]
More updates with attempts to get the bundle correct.

**Comment by chrisndeca** [comment]
Just wondering if there has been any movement on this? Would love to try and do a build on my Mac.

**Comment by loganmc10** [comment]
The problem with this PR is that it puts all the components inside "mupen64plus-gui", which is wrong. Mupen64plus-gui is just one of the components.

Unfortunately I don't have a Mac so I can't really do much here. I would love to be able to provide Mac builds on m64p.github.io though, I've had a few people request it. I just need someone to help me out with it a bit.

**Comment by fprimex** [comment]
Yes this was not complete or working at the time I PR'd it. I just wanted to get it out there more as a reference in case someone wanted to try to push it the rest of the way through. I got as far as "I can compile and run the app, but if I try to zip the app and move it to a non-development machine, it crashes saying it can't locate libraries."

So there are still things that need to be fixed with respect to the location of the libraries (which I think I eventually got right), and the modification of the binaries to be able to find those libraries using something like `install-name-tool`. I thought I finally got it using a command that comes with Qt, but that didn't work either.

We need someone with more experience packaging apps on MacOS, specifically with Qt. Or I just need more time to work on this, which has been nonexistent for several months now.

**Comment by fprimex** [comment]
Closing in favor of #8

**Pull request closed without merging**